state/factory/workingset.go

```diff
@@ -155,6 +157,15 @@ func (ws *workingSet) runAction(
 	if protocol.MustGetBlockCtx(ctx).GasLimit < protocol.MustGetActionCtx(ctx).IntrinsicGas {
 		return nil, errors.Wrap(action.ErrHitGasLimit, "block gas limit exceeded")
 	}
+	// Reject execution of chainID not equal the node's chainID
+	blkChainCtx := protocol.MustGetBlockchainCtx(ctx)
+	g := genesis.MustExtractGenesisContext(ctx)
+	if g.IsJutland(ws.height) {
```

Comment:
after https://github.com/iotexproject/iotex-core/pull/2782 merge, change to `IsKamchatka()`

Reply:
add test case in `local_transfer_test.go` using wrong chainID, and make sure it is caught and rejected here

Reply:
#2782 has merged, change to IsKamchatka()